app/controllers/interviews_controller.rb

```diff
@@ -0,0 +1,177 @@
+        format.html { redirect_to @interview, notice: 'Interview schedule was successfully updated.' }
+        format.json { render :show, status: :ok, location: @interview }
+      else
+        format.html { render :edit }
+        format.json { render json: @interview.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    @interview.destroy
+    respond_to do |format|
+      format.html { redirect_to interviews_url, notice: 'Interview schedule was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+    def set_interview
+        @interview = Interview.find(params[:id])
+    end
+
+    def interview_params
+        params.require(:interview).permit({interviews_users_attributes: [users: [:name, :email, :resume]]}, :interview_date, :start_time, :end_time)
+    end
+
+    def welcome
+        candidate_params = interview_params[:interviews_users_attributes]["1"][:users]
+
+        h = JSON.generate({ 'type' => 'welcome', 
+            'name' => candidate_params[:name],
+            'email' => candidate_params[:email],
+            'message' => "Welcome to interview scheduler!"})
+    
+        PostmanWorker.perform_async(h, 1)
+    end
+
+    def interview_create_mail
+        candidate_params = interview_params[:interviews_users_attributes]["1"][:users]
+
+        h = JSON.generate({ 'type' => 'interview_create', 
+            'name' => candidate_params[:name],
+            'email' => candidate_params[:email],
+            'date' => interview_params[:interview_date],
+            'start_time' => interview_params[:start_time],
+            'end_time' => interview_params[:end_time]})
+    
+        PostmanWorker.perform_async(h, 1)
+    end
+
+    def interview_update_mail
+        candidate_params = interview_params[:interviews_users_attributes]["1"][:users]
+
+        h = JSON.generate({ 'type' => 'interview_update', 
+            'name' => candidate_params[:name],
+            'email' => candidate_params[:email],
+            'date' => interview_params[:interview_date],
+            'start_time' => interview_params[:start_time],
+            'end_time' => interview_params[:end_time]})
+    
+        PostmanWorker.perform_async(h, 1)
+    end
+
+    def interview_reminder_mail
+        candidate_params = interview_params[:interviews_users_attributes]["1"][:users]
+
+        h = JSON.generate({ 'type' => 'reminder', 
+            'name' => candidate_params[:name],
+            'email' => candidate_params[:email],
+            'date' => interview_params[:interview_date],
+            'start_time' => interview_params[:start_time],
+            'end_time' => interview_params[:end_time],
+            'message' => '30 min reminder'})
+
+        date = interview_params[:interview_date]
+        start_time = interview_params[:start_time]
+
+        remind_at = date.to_time.to_i + start_time.to_i - 30 * 60
+
+        PostmanWorker.perform_at(remind_at, h, 1)
+    end
+end
```

Comment:
Note: Great job on not writing queries on controller. With a bit of refactoring, your code should be good to go.